### PR TITLE
PullRequest for Issue1288 Make bioxas Acquaman use new dialog and network storage

### PR DESCRIPTION
**Pull request by XuetaoLiu-CLS**
https://github.com/acquaman/acquaman/issues/1288

1) make bioxas acuqman use new file dialog
    local storage: `/AcquamanLocalData/bioxas/AcquamanSideData`, `/AcquamanLocalData/bioxas/AcquamanMainData` and `/AcquamanLocalData/bioxas/AcquamanImagingData`
   network storage: `/home/bioxas/AcuamanSideData`, `/home/bioxas/AcuamanMainData` and `/home/bioxas/AcuamanImagingData`

2) change the file dialog for easier usage

@davidChevrier @dretrex @helfrij

### CODE REVIEW

File: BioXASSideAcquaman.pro.user [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!DOCTYPE QtCreatorProject>
-<!-- Written by Qt Creator 2.3.1, 2015-02-02T14:01:24. -->
```
Comment (dretrex) [inline BioXASSideAcquaman.pro.user]: @XuetaoLiu-CLS, the .pro.user files shouldn't be part of the commit.

**Comment by dretrex** [comment]
@XuetaoLiu-CLS, I know you made a bunch of changes to the CLS network synchronizer as well, are those going to be in a separate pull request?

**Comment by XuetaoLiu-CLS** [comment]
@dretrex I'm cleaning up my changes. It looks like I had too many "thoughts" about this issue. I will try to minimize the changes and test the possible scenarios.

**Comment by dretrex** [comment]
Okay, let me know when you're in a stage where you want me to look over it.

**Comment by XuetaoLiu-CLS** [comment]
@dretrex I finished the coding. Please recheck. Thanks.

It appears that there is no need to change for the CLSNetworkSynchronizer.

**Comment by dretrex** [comment]
Hey @XuetaoLiu-CLS, this looks good to me.  The only question I have is whether you've tested this on any of the other beamlines to confirm that it truly does what we expect.  We really don't want this to have any bugs in it.

**Comment by XuetaoLiu-CLS** [comment]
@dretrex I didn't test on the other beamlines. It is only tested on BioXAS.

File: source/application/AMDatamanAppController.cpp [hunk 2]
```diff
@@ -356,16 +356,16 @@ bool AMDatamanAppController::startupOnFirstTime()
 			return false;
 	}
 	else{
-		AMFirstTimeWizard ftw;
+		AMFirstTimeWizard *firstTimeWizard = new AMFirstTimeWizard(AMUserSettings::userBasedDataStorage);
```
Comment (iainworkman) [inline source/application/AMDatamanAppController.cpp]: I don't see any parenting for the first time wizard. Will it need deleteLater() called on it at all the exit point to this block?

**Comment by iainworkman** [comment]
I pulled this and built on SGM. Everything built fine and ran. Are there anythings you want us to try out when we're testing this out @XuetaoLiu-CLS ?

File: source/application/AMDatamanAppController.cpp (hunk 2 above)
Comment (dretrex) [inline source/application/AMDatamanAppController.cpp]: Yes it will since it was changed to a pointer it will need to be cleaned up before leaving the code block.

**Comment by XuetaoLiu-CLS** [comment]
@iainworkman nothing particular, just test the application with proposal number or absolute storage path; and make sure the network synchronizer is working properly (when save locally is enabled).

**Comment by iainworkman** [comment]
Yep, everything seems fine with this down on SGM. SGM uses user based data storage though, so some of the stuff is not testable here.

**Comment by XuetaoLiu-CLS** [comment]
@iainworkman Thanks Ian. The user based data storage is the one we are not sure about.

**Comment by dretrex** [comment]
Okay, so my understanding is that we've now done all the testing we wanted?  Is this now in a state where I can merge it in?

File: source/application/AMDatamanAppController.cpp (hunk 2 above)
Comment (dretrex) [inline source/application/AMDatamanAppController.cpp]: @XuetaoLiu-CLS can you add the deleteLater() call?  I think that is the last of any comments.  I'll merge this in once you've done that.

Comment (XuetaoLiu-CLS) [inline source/application/AMDatamanAppController.cpp]: @dretrex it is added.

**Comment by dretrex** [comment]
Alright, great.  Thanks for all the work on this one @XuetaoLiu-CLS!
